llvm/lib/Transforms/Scalar/ConstraintElimination.cpp

```diff
@@ -1260,14 +1267,12 @@ static void generateReproducer(CmpInst *Cond, Module *M,
   assert(!verifyFunction(*F, &dbgs()));
 }
 
-static std::optional<bool> checkCondition(CmpInst *Cmp, ConstraintInfo &Info,
+static std::optional<bool> checkCondition(CmpInst::Predicate Pred, Value *A,
```

Comment:
Could you submit a separate PR for this NFC refactoring to reduce the diff here?